redis/db/redisbp/monitored_client.go

```diff
@@ -25,6 +26,13 @@ type PoolStatser interface {
 func NewMonitoredClient(name string, opt *redis.Options) *redis.Client {
 	client := redis.NewClient(opt)
 	client.AddHook(SpanHook{ClientName: name})
+
+	if err := prometheus.Register(newExporter(client, name)); err != nil {
+		// prometheus.Register should never fail because
+		// exporter.Describe is a no-op, but just in case.
+		return nil
```

Comment:
🔕 I personally would slightly prefer to use `MustRegiester` over returning an `nil` client (but still add a comment to explain that this is not supposed to happen).

pros of `MustRegister`: when it actually happens, the panic message will be the error returned by `Register`, so it's more obvious that it's the prometheus exporter causing trouble. vs. returning `nil`, which likely will also cause panic, but the panic message would be about nil dereference, and it's much harder to figure out why.

pros of using `nil`: caller can actually check for `nil` and fail more gracefully than just panic.

Reply:
Yes, that's why I chose to just return nil, it's easy to check.